drivers/power/ltc3350/ltc3350.c

```diff
@@ -0,0 +1,318 @@
+* LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include "no_os_util.h"
+#include "no_os_delay.h"
+#include "no_os_alloc.h"
+#include "no_os_i2c.h"
+
+#include "ltc3350.h"
+
+/***************************************************************************//**
+ * @brief Initializes the communication peripheral and checks if the LTC3350
+ *        part is present.
+ *
+ * @param device     - The device structure.
+ * @param init_param - The structure that contains the device initial
+ * 		       parameters.
+ *
+ * @return ret       - Result of the initialization procedure.
+*******************************************************************************/
+int ltc3350_init(struct ltc3350_dev **device,
+		 struct ltc3350_init_param *init_param)
+{
+	struct ltc3350_dev *dev;
+	int ret;
+	dev = (struct ltc3350_dev *)no_os_calloc(1, sizeof(*dev));
+	if (!dev)
+		return -ENOMEM;
+
+	ret = no_os_i2c_init(&dev->i2c_desc, &init_param->i2c_init);
+	if (ret) {
+		goto err;
+	}
+
+	return 0;
+
+err:
+	no_os_free(dev);
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Free the resources allocated by ltc3350_init().
+ *
+ * @param dev - The device structure.
+ *
+ * @return ret - Result of the remove procedure.
+*******************************************************************************/
+int ltc3350_remove(struct ltc3350_dev *dev)
+{
+	int ret;
+
+	ret = no_os_i2c_remove(dev->i2c_desc);
+	no_os_free(dev);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Reads data from the device (I2C).
+ *
+ * @param dev          - The device structure.
+ * @param base_address - Address of the base register.
+ * @param read_data    - The read data value
+ *
+ * @return ret         - Result of the reading procedure.
+*******************************************************************************/
+int ltc3350_read_device_data(struct ltc3350_dev *dev, uint8_t base_address,
+			     uint16_t *read_data)
+{
+	int ret;
+	uint8_t rx_buff[2] = {0};
+
+	ret = no_os_i2c_write(dev->i2c_desc, &base_address, 1, 1);
+	if (ret)
+		return ret;
+
+	ret = no_os_i2c_read(dev->i2c_desc, rx_buff, 2, 1);
+	if (ret) {
```

Comment:
why setting *read_data to 0 if the error code will be treated anyways in the upper layer?